src/services/userService.js

```diff
@@ -39,11 +39,32 @@ async function login(username, password) {
     }
 }
 
-async function getUserByUsername(username) {
-    const result = await userDAO.queryByUsername(username);
+const updateRole = async (id, role) => {
+    let result = await userDAO.getUserById(id);
```

Comment:
I understand that you use let result so you can reassign it to the new result after updating role. However, it can be confusing to have the response to both be the same variable (though this is just my opinion).

Reply:
You're right, I can separate them into different variables.